components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java

```diff
@@ -252,4 +252,34 @@ InputStream getFileById(String resourceType, String resourceName, String fileId)
     void deleteFileById(String resourceType, String resourceName, String fileId)
             throws ConfigurationManagementException;
 
+    /**
+     * This function is used to get a resource by the resource id.
+     *
+     * @param resourceId Id representing the resource.
+     * @throws ConfigurationManagementException Configuration management exception.
+     */
+    default Resource getTenantResourceById(String resourceId) throws ConfigurationManagementException {
+
+        return null;
```

Comment:
Shall we throw the `NotImplementedException` in Apache commons library?

Reply:
We have implemented this method in the child class, `ConfigurationManagerImpl.java` where this interface is implemented. Therefore we don't need to have default methods at this level.